File: CodeForces/678_ProbB.py
from sys import stdin,stdout
stdin = open("/Users/seeva92/Workspace/Contests/1.txt","r")
stdout = open("/Users/seeva92/Workspace/Contests/2.txt","w")
from functools import reduce
from operator import mul
import datetime
import calendar
def gcd(a,b):
	if b==0:
		return a
	return gcd(b,a%b)
def main():
	stdout.write(str(gcd(15,30)))
    # stdout.write("wow")
    # year = int(stdin.readline().strip())
    # first = calendar.weekday(year,1,1)
    # for year in range(1000,2051):
        # first = calendar.weekday(year,1,1)
        # for y in range(year+1,100001):
            # if first == calendar.weekday(y,1,1) and calendar.isleap(first) == calendar.isleap(y):
                # stdout.write(str(y)+'\n'); break
    # first = datetime.datetime(2016,6,13)
    # second = datetime.datetime(2017,6,13)
    # stdout.write(str(calendar.months[calendar.weekday(2017,6,13)]))
    # stdout.write(str(365%7))
    # stdout.write(str((second-first).days))
main()

File: CodeForces/678_ProbB.cpp
#include <climits>
#include <cstdio>
#include <iostream>
#include <unordered_map>
#define vi vector<int>
#define all(v) v.begin(), v.end()
#define pii pair<int, int>
#define mp make_pair
#define ll long long
using namespace std;
inline bool get(int temp) {
  if ((temp % 400 == 0) || ((temp % 4 == 0) && (temp % 100 != 0)))
    return true;
  return false;
}
int main() {
#ifndef ONLINE_JUDGE
  freopen("/Users/seeva92/Workspace/Contests/1.txt", "r", stdin);
  freopen("/Users/seeva92/Workspace/Contests/2.txt", "w", stdout);
#endif
  ios::sync_with_stdio(false);
  cin.tie(0);
  int n;
  cin >> n;
  int val = 1;
  for (int year = 2016; year > 1000; year--) {
    if (get(year)) {
      val = (7 + val - (366 % 7)) % 7;
    } else {
      val = (7 + val - (365 % 7)) % 7;
    }
  }
  for (int year = 1001; year <= n; year++) {
    val += get(year) + 365;
    val %= 7;
  }
  int res = val;
  int isleap = get(n);
  int j = n + 1;
  while (1) {
    val += get(j) + 365;
    val %= 7;
    if (val == res && isleap == get(j)) {
      cout << j << endl;
      break;
    }
    ++j;
  }
}
